crates/ide_db/src/helpers.rs

```diff
@@ -230,11 +230,11 @@ pub fn for_each_tail_expr(expr: &ast::Expr, cb: &mut dyn FnMut(&ast::Expr)) {
     }
 }
 
-/// Calls `cb` on each break expr inside of `body` that is applicable for the given label.
-pub fn for_each_break_expr(
+/// Calls `cb` on each break expr and continue expr inside of `body` that is applicable for the given label.
+pub fn for_each_break_and_continue_expr(
```

Comment:
Given my review above, we probably have to duplicate this somewhat to have one for only break expressions and one with both break and continue

Reply:
I addressed your points. I decided to share logic by wrapping the WalkEvent + depth logic that is done here within an iterator. That way those two functions can be done without sharing too much code.

Thanks for you review!